### PR TITLE
parameters panel, showing the configuration file contents

### PR DESCRIPTION
**Pull request by e3rd**
Monitor tab gets parameters panel that displays bot parameters. If there is a file or a directory from within the allowed path (specified in configs.php), the contents gets displayed so the administrator is aware about the configuration the bot has.

What do you think?

### CODE REVIEW

File: intelmq-manager/js/monitor.js [hunk 1]
```diff
@@ -316,6 +329,32 @@ function select_bot(bot_id, history_push = false) {
...
+        return;
+    }
+    let $panel = $("#parameters-panel .panel-body");
+    $panel.text("");
+    var params = app.nodes[bot_id].parameters;
+    console.log(params);
+    for (let key in params) {
+        $el = $("<li><b>"+key+"</b>: "+params[key]+"</li>");
+        console.log(params[key],key);
+        if (params[key].indexOf && params[key].indexOf(ALLOWED_PATH) === 0) {
```
Comment (wagner-certat) [inline intelmq-manager/js/monitor.js]: In the console I see this error message:
```
TypeError: params[key] is null       monitor.js:345:1
refresh_parameters_panel             http://localhost/js/monitor.js:345:1
select_bot                                     http://localhost/js/monitor.js:320:9
select_bot_func/<                          http://localhost/js/monitor.js:378:21
```
I think it is related to this line.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ CHANGELOG
 
 #### Monitor
 * Fix link to monitor page (#157).
+* Parameters panel that displays bot parameters. If there is a file or a directory from within the allowed path (specified in configs.php), the contents gets displayed so the administrator is aware about the configuration the bot has.
```
Comment (wagner-certat) [inline CHANGELOG.md]: I needed some time to get the meaning of the allowed paths. Proposal:
> The new parameters panel that displays the bot's parameters on the bot's monitor page. For parameters which are filenames, the content of these files will be shown as text if they reside in the allowed paths (set in config.php).

**Review by ghost — CHANGES_REQUESTED** [review]
If no bot is selected, the box's content is just "loading...".

If the file does not exist the shown file *content* is "Unknown resource".

Loading files is not always a good idea:
1. binary data. It will be shown without error, but it's not useful.
2. big files, crashes the browser in the worst case.

Suggestion: Show a link to display files on demand. And/or show the file size beforehand and/or load it automatically if small.

File: intelmq-manager/js/monitor.js (hunk 1 above)
Comment (e3rd) [reply intelmq-manager/js/monitor.js]: I hope I've corrected this.

File: CHANGELOG.md (hunk 2 above)
Comment (e3rd) [reply CHANGELOG.md]: fully agree

**Comment by e3rd** [comment]
I've implemented the fetching mechanism for the files (by default) bigger than 2 kB.

> If no bot is selected, the box's content is just "loading...".

Done.

> If the file does not exist the shown file content is "Unknown resource".

This feature is surprisingly hard to achieve. In order to prevent the directory traversal, I cast realpath on the filename wanted. If the filename does not exist, I receive the empty string. If I receive the empty string I cannot tell if the user had wanted a file in an allowed path which does not exist or if they were trying to trick me with a path containing something like "/opt/intelmq/allowed-path/../../../etc/" to check what files exist on the file system. What if I chagne "Unknown resource" to "Resource not reachable"?

**Review by ghost — APPROVED** [review]

**Pull request merged**